### ISSUE TITLE
Corefinement Intersection Error

### ISSUE THREAD

**Issue by JamesYangLim** [CGAL/cgal#2956]
Hi @sloriot,

I have tried your fix to the [corefinement intersection issue I recently submitted](https://github.com/CGAL/cgal/issues/2939) however I'm now getting a different assertion in the same file (intersection_of_coplanar_triangles_3.h), while doing a similar intersection calculation, as shown in the diagram below.

```
CGAL error: assertion violation!
Expression : ipt_prev.type_1!=ON_FACE
File       : c:\vcpkg\installed\x64-windows\include\cgal\polygon_mesh_processing\internal\corefinement\intersection_of_coplanar_triangles_3.h
Line       : 130
Explanation:
Refer to the bug-reporting instructions at http://www.cgal.org/bug_report.html
unknown file: error: C++ exception with description "CGAL ERROR: assertion violation!
Expr: ipt_prev.type_1!=ON_FACE
File: c:\vcpkg\installed\x64-windows\include\cgal\polygon_mesh_processing\internal\corefinement\intersection_of_coplanar_triangles_3.h
Line: 130" thrown in the test body.
```

![image](https://user-images.githubusercontent.com/30457655/37891736-f158f69c-3107-11e8-9fc1-7a7876059d15.png)

## Source Code

```
 using Mesh = CGAL::Surface_mesh<typename CGAL::Epeck::Point_3>;
    std::string bigBox = 
        R"(OFF
8 12 0
360 4950 50
450 4950 50
360 5040 50
450 5040 50
360 4950 100
450 4950 100
360 5040 100
450 5040 100
3 0 2 3
3 3 1 0
3 4 5 7
3 7 6 4
3 0 1 5
3 5 4 0
3 3 2 7
3 2 6 7
3 1 3 5
3 3 7 5
3 0 4 2
3 4 6 2
)";

    std::string smallBox =
        R"(OFF
8 12 0
375 4925 85
400 4925 85
375 4950 85
400 4950 85
375 4925 90
400 4925 90
375 4950 90
400 4950 90
3 0 2 3
3 3 1 0
3 4 5 7
3 7 6 4
3 0 1 5
3 5 4 0
3 3 2 7
3 2 6 7
3 1 3 5
3 3 7 5
3 0 4 2
3 4 6 2
)";

    std::istringstream iss(bigBox);
    Mesh bigBoxMesh;
    iss >> bigBoxMesh;

    std::istringstream iss2(smallBox);
    Mesh smallBoxMesh;
    iss2 >> smallBoxMesh;

    auto intersect = CGAL::Polygon_mesh_processing::corefine_and_compute_intersection(smallBoxMesh, bigBoxMesh, smallBoxMesh);
```

## Environment

* Operating system (Windows 64 bits):
* Compiler: C++
* Release or debug mode: Release
* CGAL version: v4.11
* Boost version: 1.66.0

**Comment by sloriot** [comment]
Thanks for the report. #2958 should fix the assertion.
Note that by convention, I decided that the intersection polygon will be empty since the volume will be empty.

**Comment by lrineau** [comment]
Fixed by #2958.
